src/main/java/com/ericsson/ei/queryservice/ProcessFailedNotification.java

```diff
@@ -47,17 +47,17 @@
     private MongoDBHandler handler;
 
     /**
-     * The method is responsible to extract the data on the basis of the
-     * subscriptionName from the Missed Notification Object.
+     * Get a failed notification object by using the subscriptionName key.
      *
      * @param subscriptionName
      * @return List
      */
-    public List<String> processQueryMissedNotification(String subscriptionName) {
+    public List<String> processQueryFailedNotification(String subscriptionName) {
```

Comment:
I think this is very unclear as an interface. If we take only a string then I would believe that it's only one element you can send in.

Id rather put it as a "post" and put a list in a json body or something. Creating our own format in a string is not suitable in a rest interface in my mind. But ye, this would require changes in other places as well since we do it in other places.

Reply:
As for now, it is. We only search for documents that have the given subscriptionName

Reply:
This is only a suggestion for the REST api not what happens in the logic after. However, I talked with Christoffer about this and we will keep as is since we have the same structure in other endpoints and this is what we agreed in #291

Reply:
It IS only one element that you send in *laughs* Look at Anders comment above about why we don't fetch all notifications in a single query. This class was written in such a way to fetch a single one at a time. This is also how subscriptions work... it really seems like a lot of overhead. If you type 10 subscriptions then you have to make 10 connections and queries instead of a single one.

Reply:
I commented in the wrong place, ignore my comment. Its about the REST api ;)